tests/sentry/utils/performance_issues/test_slow_span_detector.py

```diff
@@ -0,0 +1,105 @@
+from sentry.testutils.silo import region_silo_test
+from sentry.utils.performance_issues.performance_detection import (
+    DetectorType,
+    GroupType,
+    PerformanceProblem,
+    SlowSpanDetector,
+    get_detection_settings,
+    prepare_problem_for_grouping,
+    run_detector_on_data,
+)
+
+
+@region_silo_test
+@pytest.mark.django_db
+class SlowSpanDetectorTest(unittest.TestCase):
+    def setUp(self):
+        super().setUp()
+        self.settings = get_detection_settings()
+
+    def find_slow_span_problems(self, event: Event) -> List[PerformanceProblem]:
+        detector = SlowSpanDetector(self.settings, event)
+        run_detector_on_data(detector, event)
+        # The usage of "prepare_problem_for_grouping" is only needed because the slow db detector is using PerformanceSpanProblem,
+        # TODO - refactor slow span detector to use PerformanceProblem
+        return [
+            prepare_problem_for_grouping(problem, event, DetectorType.SLOW_SPAN)
+            for problem in detector.stored_problems.values()
+        ]
+
+    def test_calls_detect_slow_span(self):
+        no_slow_span_event = create_event([create_span("db", 999.0)] * 1)
+        slow_not_allowed_op_span_event = create_event([create_span("random", 1001.0, "example")])
+        slow_span_event = create_event([create_span("db", 1001.0)] * 1)
+
+        assert self.find_slow_span_problems(no_slow_span_event) == []
+        assert self.find_slow_span_problems(slow_not_allowed_op_span_event) == []
+        assert self.find_slow_span_problems(slow_span_event) == [
+            PerformanceProblem(
+                fingerprint="1-GroupType.PERFORMANCE_SLOW_SPAN-020e34d374ab4b5cd00a6a1b4f76f325209f7263",
+                op="db",
+                desc="SELECT count() FROM table WHERE id = %s",
+                type=GroupType.PERFORMANCE_SLOW_SPAN,
+                parent_span_ids=None,
+                cause_span_ids=None,
+                offender_span_ids=["bbbbbbbbbbbbbbbb"],
+            )
+        ]
+
+    def test_calls_slow_span_threshold(self):
+        http_span_event = create_event(
+            [create_span("http.client", 1001.0, "http://example.com")] * 1
+        )
+        db_span_event = create_event([create_span("db.query", 1001.0)] * 1)
+
+        assert self.find_slow_span_problems(http_span_event) == []
+        assert self.find_slow_span_problems(db_span_event) == [
+            PerformanceProblem(
+                fingerprint="1-GroupType.PERFORMANCE_SLOW_SPAN-dff8b4c2cafa12af9b5a35f3b12f9f8c2d790170",
+                op="db.query",
+                desc="SELECT count() FROM table WHERE id = %s",
+                type=GroupType.PERFORMANCE_SLOW_SPAN,
+                parent_span_ids=None,
+                cause_span_ids=None,
+                offender_span_ids=["bbbbbbbbbbbbbbbb"],
+            )
+        ]
+
+    def test_calls_partial_span_op_allowed(self):
```

Comment:
You're already basically testing this in the test above, aren't you? Maybe make the op in the test above just `db` instead. Just a nit tho

Reply:
Or we could delete this test instead since its code path is already being tested by the test above

Reply:
Yeah i'll delete it, I just copied these test cases over while doing this refactor, but its a good idea to cleanup while at it.

Reply:
right right, I should've said "we're" instead 😄

Reply:
Deleted!